src/core/imap/MCIMAPSession.cpp

```diff
@@ -3646,6 +3646,7 @@ void IMAPSession::storeFlagsAndCustomFlags(String * folder, bool identifier_is_u
             return;
         }
     }
+    * pError = ErrorNone;
```

Comment:
Same here: Could change the "return"s above to "goto release" too? It would be a great mem leak fix :)